pkg/services/authz/rbac/service.go

```diff
@@ -140,12 +207,19 @@ func (s *Service) getUserPermissions(ctx context.Context, req *CheckRequest) ([]
 		return nil, fmt.Errorf("could not get user internal id: %w", err)
 	}
 
-	basicRoles, err := s.store.GetBasicRoles(ctx, req.Namespace, store.BasicRoleQuery{UserID: userIdentifiers.ID})
-	if err != nil {
-		return nil, fmt.Errorf("could not get basic roles: %w", err)
-	}
+	s.idCache.Set(uidCacheKey, userIdentifiers, 0)
```

Comment:
Not really best practice, but as we are storing twice, I'm storing a pointer. Wdyt?